Networking/Networking/Model/Product/Product.swift

```diff
@@ -109,6 +109,13 @@ public struct Product: Codable, GeneratedCopiable, Equatable {
         return ProductTaxStatus(rawValue: taxStatusKey)
     }
 
+    /// Whether the product variation has an integer stock quantity.
+    /// Decimal (non-integer) stock quantities currently aren't accepted by the Core API.
+    /// Related issue: https://github.com/woocommerce/woocommerce-ios/issues/3494
+    public var hasIntegerStockQuantity: Bool {
+        stockQuantity?.exponent ?? 0 >= 0
```

Comment:
nit: TIL'ed [`Double.exponent`](https://developer.apple.com/documentation/swift/double/1848471-exponent) though I'm not sure if I fully understand how this is used to check if a double is an integer 😅  do you think we could use a potentially easier solution like [this SO answer](https://stackoverflow.com/a/28448059/9185596)?

To reuse the logic with `ProductVariation`, we could also make an extension for `Double` like `Double.isInteger: Bool`?

Reply:
I chose this approach because `stockQuantity` is a `Decimal` type so we can check the exponent directly (versus converting it to a `Double` to do other math on it). It's based on [floating point arithmetic](https://en.wikipedia.org/wiki/Floating-point_arithmetic), where a negative exponent essentially means you're moving the decimal point to the left (to create a non-integer). For example, the exponent `-3` here:

> 12345 * 10^-3 = 12.345

Whereas an integer has a 0 or positive exponent:

> 12345 * 10^0 = 12345

or

> 12345 * 10^3 = 12345000

Do you think it'd make sense to use this with a comment explaining the math behind it? If that still seems like it'd be unclear, I can switch it out for a solution like the one you found.

Either way, great idea to make an extension to reuse the logic — I'll work on that!

Reply:
After thinking this over, I moved the existing `exponent` check into a `Decimal.isInteger` extension in https://github.com/woocommerce/woocommerce-ios/pull/3717/commits/0ea9a52a207202ebe91a25720d16dd6bb47ce8e9. I added an explanation there of how the arithmetic works, which hopefully should help with clarity. Let me know if you think it could still be improved!

I also added a unit test in https://github.com/woocommerce/woocommerce-ios/pull/3717/commits/5b522c9790debb448d1679d59901dec480d816a2 to confirm it works with a variety of potential values.

Reply:
The explanation you added is super helpful and easy to understand! I was confused by the example in [the Apple doc](https://developer.apple.com/documentation/swift/double/1848471-exponent) where the radix is 2 😅 

The extension changes look great with a unit test! 💯